lib/sidebar.rb

```diff
@@ -46,93 +66,96 @@ def post_write(site)
           }
           headers.push(header)
         end
-        sanitized_filename = sanitize_filename(filename)
+
         @filename_with_headers[sanitized_filename] = { headers: headers }
       end
 
-      Dir.glob("#{site.config['destination']}/**/*.html") do |filename|
-        sanitized_filename = sanitize_filename(filename)
+      files
+    end
+
+    def write_sidebar_to_files(files)
+      files.each do |file|
+        sanitized_filename = file[:sanitized_filename]
+        filename = file[:filename]
+        doc = file[:doc]
         sidebar = render(sanitized_filename)
-        file = File.open(filename) { |f| Nokogiri::HTML(f) }
-        file.xpath('//*[@id="dx-sidebar-main-nav-ul"]').each do |location|
-          location.inner_html = sidebar
+
+        doc.xpath('//*[@id="dx-sidebar-main-nav-ul"]').each do |sidebar_container|
+          sidebar_container.inner_html = sidebar
         end
-        File.open(filename, 'w') { |f| f.write(file.to_html(encoding: 'UTF-8')) }
-      end
 
-      # File.open('_site/sidebar.html', 'w') { |f| f.write(sidebar) }
+        File.open(filename, 'w') { |f| f.write(doc.to_html(encoding: 'UTF-8')) }
+      end
     end
 
-    private
+    def generate_sub_group(filename, key, value, all_sub_groups, level)
```

Comment:
Codacy found an issue: [Cyclomatic complexity for generate_sub_group is too high. [12/6]](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=6198407)